SU2_CFD/src/solver_direct_turbulent.cpp

```diff
@@ -2952,6 +2966,279 @@ void CTurbSASolver::BC_NearField_Boundary(CGeometry *geometry, CSolver **solver_
+    iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+    
+    /*--- We can use also GetNormal_Neighbor, and eliminate the following loop ---*/
+    
+    iPoint_Neighbor = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+    for(iNode = 0; iNode < geometry->node[iPoint]->GetnPoint(); iNode++) {
+      iPoint_Neighbor = geometry->node[iPoint]->GetPoint(iNode);
+      
+      /*--- Check if the node belongs to the domain (i.e, not a halo node)
+       and the neighbor is not part of the physical boundary ---*/
+      
+      if (geometry->node[iPoint]->GetDomain() && (!geometry->node[iPoint_Neighbor]->GetBoundary())) {
+        
+        /*--- Get coordinates of the current vertex and nearest normal point ---*/
+        
+        Coord = geometry->node[iPoint]->GetCoord();
+        Coord_Normal = geometry->node[iPoint_Neighbor]->GetCoord();
+        
+        /*--- Compute dual-grid area and boundary normal ---*/
+        
+        Normal = geometry->vertex[val_marker][iVertex]->GetNormal();
+        
+        Area = 0.0;
+        for (iDim = 0; iDim < nDim; iDim++)
+          Area += Normal[iDim]*Normal[iDim];
+        Area = sqrt (Area);
+        
+        for (iDim = 0; iDim < nDim; iDim++)
+          UnitNormal[iDim] = -Normal[iDim]/Area;
+        
+        /*--- Get the velocity, pressure, and temperature at the nearest
+         (normal) interior point. ---*/
+        
+        for (iDim = 0; iDim < nDim; iDim++)
+          Vel[iDim]    = solver_container[FLOW_SOL]->node[iPoint_Neighbor]->GetVelocity(iDim);
+        P_Normal       = solver_container[FLOW_SOL]->node[iPoint_Neighbor]->GetPressure();
+        T_Normal       = solver_container[FLOW_SOL]->node[iPoint_Neighbor]->GetTemperature();
+
+        /*--- Compute the wall-parallel velocity at first point off the wall ---*/
+        
+        VelNormal = 0.0;
+        for (iDim = 0; iDim < nDim; iDim++)
+          VelNormal += Vel[iDim] * UnitNormal[iDim];
+        for (iDim = 0; iDim < nDim; iDim++)
+          VelTang[iDim] = Vel[iDim] - VelNormal*UnitNormal[iDim];
+        
+        VelTangMod = 0.0;
+        for (iDim = 0; iDim < nDim; iDim++)
+          VelTangMod += VelTang[iDim]*VelTang[iDim];
+        VelTangMod = sqrt(VelTangMod);
+        
+        /*--- Compute normal distance of the interior point from the wall ---*/
+        
+        for (iDim = 0; iDim < nDim; iDim++)
+          WallDist[iDim] = (Coord[iDim] - Coord_Normal[iDim]);
+        
+        WallDistMod = 0.0;
+        for (iDim = 0; iDim < nDim; iDim++)
+          WallDistMod += WallDist[iDim]*WallDist[iDim];
+        WallDistMod = sqrt(WallDistMod);
+        
+        /*--- Compute mach number ---*/
+        
+        M_Normal = VelTangMod / sqrt(Gamma * Gas_Constant * T_Normal);
+        
+        /*--- Compute the wall temperature using the Crocco-Buseman equation ---*/
+        
+        T_Wall = T_Normal * (1.0 + 0.5*Gamma_Minus_One*Recovery*M_Normal*M_Normal);
+        
```

Comment:
I think it should be:
T_Wall = T_Normal + Recovery*pow(VelTangMod,2.0)/(2.0*Cp);
Following Eq(7) in Eq(10).

Reply:
@economon 
I will add a regression test case similar to the paper once we agree with the modifications.